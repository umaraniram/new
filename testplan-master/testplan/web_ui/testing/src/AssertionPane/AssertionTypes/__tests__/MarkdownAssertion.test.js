import React from 'react';
import {shallow} from 'enzyme';
import {StyleSheetTestUtils} from "aphrodite";

import MarkdownAssertion from '../MarkdownAssertion';

function defaultProps() {
  return {
    assertion: {
      "category": "DEFAULT",
      "machine_time": "2019-02-12T17:41:43.302500+00:00",
      "description": null,
      "message": "Markdown Test [Github](https://github.com/Morgan-Stanley/testplan)\n Test",
      "line_no": 123,
      "meta_type": "entry",
      "type": "Markdown",
      "utc_time": "2019-02-12T17:41:43.302494+00:00"
    }
  };
}


describe('MarkdownAssertion', () => {
  let props;
  let shallowComponent;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = defaultProps();
    shallowComponent = undefined;
  });

  it('shallow renders the correct HTML structure', () => {
    shallowComponent = shallow(<MarkdownAssertion {...props} />);
    expect(shallowComponent).toMatchSnapshot();
  });
});
