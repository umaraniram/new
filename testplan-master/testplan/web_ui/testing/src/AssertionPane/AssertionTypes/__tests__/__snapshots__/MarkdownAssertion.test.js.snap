// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownAssertion shallow renders the correct HTML structure 1`] = `
<ReactMarkdown
  astPlugins={Array []}
  escapeHtml={true}
  parserOptions={Object {}}
  plugins={Array []}
  rawSourcePos={false}
  renderers={
    Object {
      "code": [Function],
    }
  }
  skipHtml={false}
  source="Markdown Test [Github](https://github.com/Morgan-Stanley/testplan)
 Test"
  sourcePos={false}
  transformLinkUri={[Function]}
/>
`;
